packages/engine/src/datastore/schema/field_spec/creator.rs

```diff
@@ -2,17 +2,57 @@ use crate::datastore::schema::field_spec::{
     FieldScope, FieldSource, FieldSpec, FieldType, RootFieldSpec,
 };
 
-/// A factory-like object that can be set with a FieldSource and then passed to a context such as a
-/// package. This allows packages to not need to be aware of the FieldSource specifics
+/// Factory for creating `RootFieldSpec`s for a given `FieldSource`.
```

Comment:
Open to discussion on this point, but, the previous docstring gave contextual reasoning for why we even have a factory-like object. I'm not super happy with us removing the reasoning and not putting it somewhere else. The Creator actually adds a lot of code and feels rather clumsy, but the reason we want it is so that packages don't need to know about FieldSources. In your docstrings you've set them to Engine, which is the easiest one because that's just a normal enum variant but when you try setting the source to a Package you need to know a lot more information, which is the logic we didn't want packages to have to deal with, so we pass them a factory object that already has that field preset for them.

Also I wrote factory-like because for some reason we have a convention of calling things `Creator` rather than `Factory` even though it really is the latter, and I felt like just straight up calling it a factory in the docstring was somewhat inconsistent.